packages/@n8n/api-types/src/dto/user/users-list-filter.dto.ts

```diff
@@ -1,25 +1,137 @@
+import { jsonParse } from 'n8n-workflow';
 import { z } from 'zod';
 import { Z } from 'zod-class';
 
 import { paginationSchema } from '../pagination/pagination.dto';
 
 const USERS_LIST_SORT_OPTIONS = [
-	'name:asc',
-	'name:desc',
+	'firstName:asc',
+	'firstName:desc',
+	'lastName:asc',
+	'lastName:desc',
 	'role:asc', // ascending order by role is Owner, Admin, Member
 	'role:desc',
-	'lastActive:asc',
-	'lastActive:desc',
+	// 'lastActive:asc',
+	// 'lastActive:desc',
 ] as const;
 
+const createTakeValidator = (maxItems: number) =>
+	z
+		.string()
+		.optional()
+		.transform((val) => (val ? parseInt(val, 10) : 10))
+		.refine((val) => !isNaN(val) && Number.isInteger(val), {
+			message: 'Param `take` must be a valid integer',
+		})
+		.transform((val) => Math.min(val, maxItems));
+
 const usersListSortByValidator = z
 	.enum(USERS_LIST_SORT_OPTIONS, {
 		message: `sortBy must be one of: ${USERS_LIST_SORT_OPTIONS.join(', ')}`,
 	})
 	.optional();
 
+const userSelectSchema = z.array(
+	z.enum(['id', 'firstName', 'lastName', 'email', 'disabled', 'mfaEnabled', 'role']),
+);
+
+const selectValidatorSchema = z
+	.string()
+	.optional()
+	.transform((val, ctx) => {
```

Comment:
I think if you send select[]=id&select[]=firstName ..., the framework is supposed to do all that for you. userSelectSchema should be enough :thinking: